### DISCUSSION TITLE
documentation?

### DISCUSSION THREAD

**Discussion by matroll** [khoih-prog/ESP8266TimerInterrupt#15] [Q&A]
thanks for the great library.
is there a documentation? anything outside of ,setInterval & .atthatchInterruptInterval?

- can you do several ITimers?
- do the ISR need ITimer or can work without?
- can the ISR be set in microseconds? or millis only?

(i need 3 independant timers set in micros)
thanks!

**Comment by khoih-prog** [comment]
Check the new [**ESP8266TimerInterrupt releases v1.4.1**](https://github.com/khoih-prog/ESP8266TimerInterrupt/releases/tag/v1.4.1)

The reason there is normally no detailed instructions for TimerInterrupt-related libraries is that it's expected that ISR users has much knowledge in hardware / firmware and will be able to read / test / understand the library / examples' code then use it without any issue.

Anyway, I already add some more instructions in the [**Usage** section](https://github.com/khoih-prog/ESP8266TimerInterrupt#usage)


For your questions

> - can you do several ITimers?

ESP8266 has just 1 timer free to use (Timer1)

> - do the ISR need ITimer or can work without?

ISR must be used with ITimer. Have a look at examples

> - can the ISR be set in microseconds? or millis only?

Interval is in microsecs, so you can use either microsecs or millis. Just remember to convert millis to microsecs by `TIMER_INTERVAL_MS * 1000`

> i need 3 independant timers set in micros

Depending on how fast the timers are. There are just pseudo-hardware ISR-timers, in the range of 100uS is possibly OK. Too small (several uS) is not possible. To be tested to be sure how small you can go and can do something useful, depending on CPU clock.

---

### Releases v1.4.1

1. Examples modified and tested with core v3.0.2
2. Add instructions in `README.md`
3. Add `changelog.md`
4. Delete Blynk-related examples

**Comment by matroll** [comment]
thank you very much for your response khoih. 
i might look into switching to an esp32 then. 
thanks


On Nov 23, 2021, at 1:26 AM, Khoi Hoang ***@***.***> wrote:

﻿
Check the new ESP8266TimerInterrupt releases v1.4.1

The reason there is normally no detailed instructions for TimerInterrupt-related libraries is that it's expected that ISR users has much knowledge in hardware / firmware and will be able to read / test / understand the library / examples' code then use it without any issue.

Anyway, I already add some more instructions in the Usage section

For your questions

can you do several ITimers?
ESP8266 has just 1 timer free to use (Timer1)

do the ISR need ITimer or can work without?
ISR must be used with ITimer. Have a look at examples

can the ISR be set in microseconds? or millis only?
Interval is in microsecs, so you can use either microsecs or millis. Just remember to convert millis to microsecs by TIMER_INTERVAL_MS * 1000

i need 3 independant timers set in micros

Depending on how fast the timers are. There are just pseudo-hardware ISR-timers, in the range of 100uS is possibly OK. Too small (several uS) is not possible. To be tested to be sure how small you can go and can do something useful, depending on CPU clock.

Releases v1.4.1

Examples modified and tested with core v3.0.2
Add instructions in README.md
Add changelog.md
Delete Blynk-related examples
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
